File: sorting/bubble-sort/bubble-sort.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Algorithms
{
    class BubbleSort {

        public static int[] BubbleSort(int[] input)
        {
            bool has_swapped = true;
            int iterations = 0, length = input.Length;

            while (has_swapped)
            {
                has_swapped = false;
                for (int i = 0; i < length-iterations-1; ++i)
                {
                    if (input[i] > input[i+1])
                    {
                        (input[i], input[i+1]) = (input[i+1], input[i]);
                        has_swapped = true;
                    }
                }
                iterations++;
            }
            return input;
        }

        static void Main(string[] args)
        {
            Console.WriteLine("Bubble Sort Test");
 
            var input = new int[] {2,5,1,2,4};
            
            Console.WriteLine("Input");
            foreach (var num in input) {
                Console.WriteLine(num);
            }
            
            var output = CountingSort(input);

            Console.WriteLine("Output");
            foreach (var num in output) {
                Console.WriteLine(num);
            }
        }
    }
}

File: sorting/bubble-sort/bubble-sort.py
def bubble_sort(input):

    has_swapped = True
    iterations = 0

    while(has_swapped):
        has_swapped = False
        for i in range(len(input) - iterations - 1):
            if input[i] > input[i+1]:
                input[i], input[i+1] = input[i+1], input[i]
                has_swapped = True
        iterations += 1
    return input

def main():
    print("Bubble Sort Test")
    input = [7,3,2,4,6]
    print(input)
    output = bubble_sort(input)
    print(output)

main()